youtube/util.py

```diff
@@ -748,29 +745,194 @@ def to_valid_filename(name):
+                    file.close()
+            except:
+                print('Unable to access ' + player_file)
+        else:
+            base_js = fetch_url(player_url, headers=headers, report_text='Fetching player url (base.js) version ' + player_version, debug_name='iframe_api_base_' + player_version + '.js')
+            try:
+                print("Saving " + player_file)
+                with open(player_file, 'wb') as file:
+                    file.write(base_js)
+                    file.close()
+            except:
+                print('Unable to access ' + player_file)
+
+        signature_timestamp = None
+        signature_timestamp_cache = settings.data_dir + '/sts_' + player_version + 'txt'
+        if require_js_player:
+            if os.path.exists(signature_timestamp_cache):
+                try:
+                    with open(signature_timestamp_cache, 'r') as file:
+                        signature_timestamp = file.read()
+                        file.close()
+                except:
+                    print('Unable to extract signature timestamp from cache')
+            else:
+                signature_timestamp = extract_signature_timestamp(base_js.decode("utf-8")).group(1)
+                try:
+                    if not os.path.exists(settings.data_dir):
+                        os.makedirs(settings.data_dir)
+                    with open(settings.data_dir + '/sts_' + player_version + '.txt', 'w') as file:
+                        file.write(signature_timestamp)
+                        file.close()
+                except:
+                    print('Unable to save signature timestamp')
+
+            if signature_timestamp != None:
+                print('Signature timestamp: ' + signature_timestamp)
+                data['playbackContext'] = {
+                        'contentPlaybackContext': {
+                          'html5Preference': 'HTML5_PREF_WANTS',
+                          'signatureTimestamp': signature_timestamp,
+                        }
+                    }
+
+    if po_token_data != None:
+        data['serviceIntegrityDimensions'] = po_token_data
+    url = 'https://' + host + '/youtubei/v1/' + api
+    if key != None:
+        url = url + '?key=' + key
+
+    print("Debugging headers")
+    for item in headers:
+        print(item)
+    print("Debugging data payload")
+    print(json.dumps(data, indent=4))
     data = json.dumps(data)
-    headers = (('Content-Type', 'application/json'),('User-Agent', user_agent))
     response = fetch_url(
         url, data=data, headers=headers,
         debug_name='youtubei_' + api + '_' + client,
         report_text='Fetched ' + client + ' youtubei ' + api
     ).decode('utf-8')
+    response_dict = json.loads(response)
+    if settings.use_visitor_data:
+        if not settings.use_po_token:
+            if response_dict['responseContext'].get('visitorData'):
+                if not os.path.exists(visitor_data_file):
```

Comment:
I added `os.path.getmtime` check to make sure that the `visitorData.txt` file is less than 86400 seconds old before using its content. Otherwise, the visitor data file will be deleted and replaced with new one.